pkg/spi-shared/config/config.go

```diff
@@ -139,17 +156,68 @@ func (c PersistedConfiguration) inflate() (Configuration, error) {
 		return conf, parseErr
 	}
 
-	if saTokenPath, ok := os.LookupEnv("SA_TOKEN_PATH"); ok {
-		conf.ServiceAccountTokenFilePath = saTokenPath
-	}
-
 	conf.KubernetesAuthAudiences = c.KubernetesAuthAudiences
 	conf.ServiceProviders = c.ServiceProviders
 	conf.SharedSecret = []byte(c.SharedSecret)
 	conf.BaseUrl = c.BaseUrl
+	conf.VaultConfiguration = c.VaultConfiguration
 	return conf, nil
 }
 
+func (c *PersistedConfiguration) applyEnv() {
+	applyTag(reflect.ValueOf(c), "env", true, func(val string) string {
+		return os.Getenv(val)
+	})
+}
+
+func (c *PersistedConfiguration) applyDefaults() {
+	applyTag(reflect.ValueOf(c), "default", false, func(val string) string {
+		return val
+	})
+}
+
+// applyTag is a helper function to PersistedConfiguration.apply* methods. It is assumed that the values are either
+// strings, arrays of strings or structs (which is true for PersistedConfiguration and its sub-structs). All other
+// types (like maps, ints, ...) are ignored.
+func applyTag(object reflect.Value, tagName string, overrides bool, valueFn func(tagValue string) string) {
```

Comment:
isn't there some lib in our dependencies that can do something like this? It doesn't look very straightforward. If not, I'd like to see unit-tests for this, to catch as much weird corner-cases from the beginning.

Reply:
Maybe we could use github.com/imdario/mergo (already in our indirect deps) or I looked at https://github.com/spf13/viper but none of those supports the definition using tags on the fields themselves. I personally like the proximity of the field and the definition of its default value and environment variable override in this approach as opposed to the code-based approach that we would have to adopt if we used either of those alternatives.